src/Compilers/Shared/DesktopBuildClient.cs

```diff
@@ -292,7 +292,35 @@ protected override int RunLocalCompilation(string[] arguments, BuildPaths buildP
                 cancellationToken.ThrowIfCancellationRequested();
 
                 Log("Attempt to connect named pipe '{0}'", pipeName);
-                pipeStream.Connect(timeoutMs);
+
+                // .NET 4.5 implementation of NamedPipeStream.Connect busy waits the entire time.
+                // Work around is to spin wait.
+                const int maxWaitInterval = 50;
+                int elapsed = 0;
+                var sw = new SpinWait();
+                while (true)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    int waitTime = Math.Min(timeoutMs - elapsed, maxWaitInterval);
+                    if (timeoutMs != Timeout.Infinite && waitTime <= 0)
+                    {
+                        Log($"Connecting to server timed out after {timeoutMs} ms");
+                        break;
```

Comment:
You might consider separating all of this new code out into its own ConnectWithSpinWait method, or something like that, which provides the same logical behavior as Connect (including the throw for timeout / cancellation / etc.) except with the added spin wait... then none of the rest of the logic needs to change, and the method could easily be swapped out if/when Connect is fixed on all supported platforms.